src/session.rs

```diff
@@ -251,12 +276,30 @@ impl <'a> Session <'a> {
   fn default_error_callback(frame : &Frame) {
     error!("ERROR received:\n{}", frame);
   }
-  
+
+  fn default_frame_send_callback(frame : &mut Frame) {
+    debug!("FRAME sending:\n{}", frame);
```

Comment:
I think the default callbacks for `on_before_send` and `on_before_receive` should be no-ops (i.e. empty).